### ISSUE TITLE
Why can't I change regex of code

### ISSUE THREAD

**Issue by BenjaminHoegh** [markedjs/marked#1269]
In the code, you use ```/^(`+)\s*([\s\S]*?[^`]?)\s*\1(?!`)/``` to check for inline code, but if you change it to ```/^(\$+)\s*([\s\S]*?[^$]?)\s*\1(?!\$)/``` its fail, why? how can this affect the code beside it should look for `$` insted of ``` ` ```

**Comment by robcresswell** [comment]
I'm not really sure this is a question for marked; looks more like a general regex question for stackoverflow or a regex matcher. It probably isn't working because the `$` needs to be escaped with `\`; in regular expressions, `$` is the termination character for the expression.

**Comment by BenjaminHoegh** [comment]
@robcresswell I know in the example its already written like that, but I only face the problem with **_marked_** I then begin to believe its have something to do with other parts of the code there mess this up, but after hours of reading the code I _haven't_ found any reason why it shouldn't work. It also has problems if you change it to other things like `%` or `BB` and that's why I choose to write a post here instead of StackOverflow.

**Comment by UziTech** [comment]
What fails? the tests?

**Comment by BenjaminHoegh** [comment]
@UziTech 
![udklip](https://user-images.githubusercontent.com/22152591/39982127-9f5dd5de-5753-11e8-9e3e-75c6c0e35453.PNG)

To be precise I do use `$$` and not just `$` but just needed an example of what's is happening

**Comment by BenjaminHoegh** [comment]
@styfle what information do you need?

**Comment by styfle** [comment]
@BenjaminHoegh This does not appear to be a problem with `marked`.

I appreciate your enthusiasm and desire to dig through the source code, but this question would be more appropriate for stackoverflow.com than on this issue tracker.

Maybe in the future we will provide a more extensible API, but that is not available at this time, and your question is about Regex, not marked.

----

That being said, the answer to your problem can be found [here](https://www.regular-expressions.info/replacecharacters.html) which discusses Special Characters and the need to escape them.

```js
var codeRegex = /^(`+)\s*([\s\S]*?[^`]?)\s*\1(?!`)/;
codeRegex.test('`code`'); // true
codeRegex.test('not code'); // false

var dollarRegex = /^(\$+)\s*([\s\S]*?[^\$]?)\s*\1(?!\$)/;
dollarRegex.test('$dollar$'); // true
dollarRegex.test('no dollars'); // false
```

Basically, the backslash can be used to escape special characters.

**Comment by BenjaminHoegh** [comment]
@styfle sure about that? because this is the result I get

    // Using:
    // code: /^(\$+)\s*([\s\S]*?[^\$]?)\s*\1(?!\$)/,
    document.getElementById('content').innerHTML = marked('inline $code$ is awesome'); // Dosn't work

    // Using:
    // code: /^(`+)\s*([\s\S]*?[^`]?)\s*\1(?!`)/,
    document.getElementById('content').innerHTML = marked('inline `code` is awesome'); // Work

**Comment by styfle** [comment]
@BenjaminHoegh Yes I’m sure. I wrote the code snippet above to prove it. 

Your example is not starting with a dollar sign so it doesn’t match. 

If you want to remove the part that only matches the beginning of a string, remove the first up-carret in the Regex. 

Read this for more info: https://www.regular-expressions.info/anchors.html

**Comment by UziTech** [comment]
@BenjaminHoegh you will have to change the `text` regex to stop at `$` instead of ` ` `

```diff
- text: /^[\s\S]+?(?=[\\<!\[`*]|\b_| {2,}\n|$)/
+ text: /^[\s\S]+?(?=[\\<!\[$*]|\b_| {2,}\n|$)/
```

**Comment by BenjaminHoegh** [comment]
@UziTech is that for block code or inline code?

----

@styfleI know, but the code there are used in the library to find code inside a paragraph use ```/^(`+)\s*([\s\S]*?[^`]?)\s*\1(?!`)/``` how this can work is a mystery for me as the `^` sign should not work when the code is inside some text, but it does. If you don't believe that the code react in a weird way and not following the normal rules of regex then try to do it in the code yourself by changing the check for ``` ` ``` to `$`

**Comment by styfle** [comment]
I was answering from my phone before so now that I'm back to a computer, I can confirm that changing those two regex works as you expect.

![regex-diff](https://user-images.githubusercontent.com/229881/40333779-380a959a-5d28-11e8-8059-af657e84d283.png)

@BenjaminHoegh Did that solve your problem?

**Comment by BenjaminHoegh** [comment]
@styfle well.. kinda works, except that its cut the text before the code... Trouble shooting here I come
